types/mithril/index.d.ts

```diff
@@ -246,7 +246,7 @@ declare namespace Mithril {
 	 * Components are a mechanism to encapsulate parts of a view to make code easier to organize and/or reuse.
 	 * Any Javascript object that has a view method is a Mithril component. Components can be consumed via the m() utility.
 	 */
-	type Comp<Attrs, State extends Lifecycle<Attrs, State>> = Component<Attrs, State> & State;
```

Comment:
One other nuance I should probably point out. This never worked with our existing types or with these changes:

```
const c: Component<{}, {}> = {
	oninit(vnode) {
		vnode.state.onremove = vnode => {};
	},
	view(vnode) {
        return m('p', 'Test')
    }
};
```

I was under the impression that it would work before I actually tested this, but it looks like the app must declare the `onremove` property in state if you want to access it via `vnode.state`. Point being we're not losing this capability with these changes. What we *do* lose is the ability to enforce that the lifecycle methods declared in a component's State type have the correct signature.